### PR TITLE
PE-4694: Credit update issue during snapshot creation purchase

### PR DESCRIPTION
**Pull request by matibat**
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/0bgop1j6s3fq8

### CODE REVIEW

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart [hunk 1]
```diff
@@ -404,6 +404,42 @@ class CreateSnapshotCubit extends Cubit<CreateSnapshotState> {
...
+    _computeIsButtonEnabled();
+
+    if (state is ConfirmingSnapshotCreation) {
+      final stateAsConfirming = state as ConfirmingSnapshotCreation;
+      logger.d('Refreshing turbo balance...');
+      logger.d('Turbo balance: $_turboCredits - ($_turboBalance)');
+      logger.d('Has no turbo balance: $_hasNoTurboBalance');
+      logger
+          .d('Sufficient balance to pay with turbo: $_sufficentCreditsBalance');
+      logger.d('Upload method: $_uploadMethod');
```
Comment (matibat) [inline lib/blocs/create_snapshot/create_snapshot_cubit.dart:431]:
```suggestion
```

**Review by thiagocarvalhodev — APPROVED** [review]
LGTM

File: lib/services/config/config_service.dart [hunk 2]
```diff
@@ -55,9 +55,10 @@ class ConfigService {
     }
   }
 
-  void updateAppConfig(AppConfig config) {
-    _configFetcher.saveConfigOnDevToolsPrefs(config);
-    _config = config;
+  void updateAppConfig(AppConfig newConfig) {
+    // logger.d('App config updated: ${config.diff(newConfig)}');
```
Comment (thiagocarvalhodev) [inline lib/services/config/config_service.dart:59]: Commented code

File: lib/turbo/topup/views/topup_modal.dart [hunk 3]
```diff
@@ -75,31 +76,39 @@ void showTurboTopupModal(BuildContext context, {Function()? onSuccess}) {
...
-    activityTracker.setToppingUp(false);
-
     if (turbo.paymentStatus == PaymentStatus.success) {
       logger.d('Turbo payment success');
 
       onSuccess?.call();
     }
 
     turbo.dispose();
+  }).whenComplete(() {
```
Comment (thiagocarvalhodev) [inline lib/turbo/topup/views/topup_modal.dart:97]: Nice catch

**Review by thiagocarvalhodev — APPROVED** [review]

File: lib/services/config/config_service.dart (hunk 2 above)
Comment (matibat) [reply lib/services/config/config_service.dart:59]: This line of code breaks a unit test, but it's also useful for when adding new options. Should I simply remove it?

**Pull request merged**